### ISSUE TITLE
[Bug]: Focus event is not fired using the launch option `headless: new`

### ISSUE THREAD

**Issue by koddsson** [puppeteer/puppeteer#10350]
### Bug expectation

I expected the `@web/test-runner-chrome` focus tests to run green after we started using the new headless option. They started failing after setting this option and the minimal reproduction posted here seems to validate that test failure.

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Minimal, reproducible example

```TypeScript
const puppeteer = require('puppeteer');

class AssertionError extends Error {}

(async () => {
  const browser = await puppeteer.launch({headless: 'new'});
  const page = await browser.newPage();

  await page.goto('https://koddsson.com');
  await page.waitForSelector('title');

  const firedEvent = await page.evaluate(async () => {
    const input = document.createElement('input');
    document.body.appendChild(input);
    
    let firedEvent = false;
    input.addEventListener('focus', () => {
      firedEvent = true;
    });
    input.focus();

    // await 2 frames for IE11
    await new Promise(requestAnimationFrame);
    await new Promise(requestAnimationFrame);
    return firedEvent;
  });

  if (firedEvent !== true) {
    throw new AssertionError('Expected focus event to have been fired!');
  }

  await browser.close();
})();
```

### Error string

AssertionError: Expected focus event to have been fired!

### Puppeteer version

20.5.0

### Node version

18.16.0

### Package manager

npm

### Package manager version

9.6.4

### Operating system

macOS

**Comment by Lightning00Blade** [comment]
Thank you for reporting. 
I can see that the same behaviors is observed with `headless: false`, I will check with the team if we should change this. 
In the mean time a workaround for anyone needing this is to use `await page.bringToFront();`.

**Comment by OrKoN** [comment]
I am not sure it is working as expected. I would expect a new page to be at the front by default. @Lightning00Blade could you report it to crbug.com?

**Comment by Lightning00Blade** [comment]
Filed https://bugs.chromium.org/p/chromium/issues/detail?id=1454012

**Comment by tmsns** [comment]
User of  `@web/test-runner` here. 👋 

We experienced similar failures when running the tests of our project. More specifically, the failures were _timeouts_. After some debugging we found that tests were waiting on code that needed to run using `window.requestAnimationFrame(callback)` and `window.requestIdleCallback(callback)`. The callbacks never seemed to be called.

Some observations we had:
- the tests did also fail when using the headfull mode (`{headless: false}`)
- the tests didn't fail with the old headless option (using `{headless: true}`)
- the tests didn't fail with the new headless option when the concurrency option of `@web/test-runner` was set to 1. This makes all the tests run in the same tab (and thus in the foreground) as opposed to multiple tabs (8).

As a workaround we override the behaviour of `window.requestAnimationFrame(callback)` and `window.requestIdleCallback(callback)` in the browser, by replacing their implementation with `window.setTimeout(callback, ms)`. In our case, this is working fine.

As the test of @koddsson is also using `requestAnimationFrame`, our problem might be related. I briefly talked to @koddsson  earlier today, and we will check if we can create a workaround in `@web/test-runner`. The idea would be to override `window.requestAnimationFrame(callback)` and `window.requestIdleCallback(callback)` in the browser as well, but then to connects back to puppeteer to run a `await page.bringToFront()` (as suggested by @Lightning00Blade). Let's hope that works 😃
